slurm_ops_manager/slurm_ops_base.py

```diff
@@ -260,11 +260,69 @@ def _slurmd_group(self) -> str:
         """Return the slurmd group."""
         return "root"
 
+    @property
+    def _slurmrestd_user_id(self) -> str:
+        """Return the slurmrestd user ID."""
+        return "64031"
+
+    @property
+    def _slurmrestd_user_name(self) -> str:
+        """Return the slurmrestd user."""
+        return "slurmrestd"
+
+    @property
+    def _slurmrestd_group_id(self) -> str:
+        """Return the slurmrestd group ID."""
+        return "64031"
+
+    @property
+    def _slurmrestd_group_name(self) -> str:
+        """Return the slurmrestd group."""
+        return "slurmrestd"
+
     @property
     def slurm_component(self) -> str:
         """Return the slurm component we are."""
         return self._slurm_component
 
+    def create_slurmrestd_user_group(self) -> None:
+        """Create the slurmrestd user."""
+
+        logger.info("#### Creating slurmrestd user and group")
+
+        try:
+            subprocess.check_output([
+                "groupadd",
+                "--gid",
+                self._slurmrestd_group_id,
+                self._slurmrestd_group_name
+            ])
+        except subprocess.CalledProcessError as e:
+            if e.returncode == 9:
+                logger.warning("## Group already exists.")
+            else:
+                logger.error(f"## Error creating group: {e}")
+                return False
+
+        try:
+            subprocess.check_output([
+                "adduser",
+                "--system",
+                "--gid", self._slurmrestd_group_id,
+                "--uid", self._slurmrestd_user_id,
+                "--no-create-home",
+                "--home", "/nonexistent",
```

Comment:
is this really correct?

Reply:
Yes, it is.

Without` --home /nonexistent`
```
$ sudo adduser --system --gid 64031 --no-create-home someuser

$ cat /etc/passwd
someuser:x:996:64031::/home/someuser:/bin/bash
```

It does not create the home dir, but it still sets it in the user info.

With `--home /nonexistent`

```
$ sudo adduser --system --gid 64031 --no-create-home --home /nonexistent someuser2

$ cat /etc/passwd
someuser2:x:995:64031::/nonexistent:/bin/bash
```

It enforces using the /nonexistent in the user info. 

adduser --system should default to /nonexistent, but it isn't.

According to https://www.debian.org/doc/debian-policy/ch-opersys.html#non-existent-home-directories, the canonical non-existent home directory is /nonexistent. Users who should not have a home directory should have their home directory set to this value.